File: 189. Rotate Array.java

import java.lang.reflect.Array;
import java.util.Arrays;

class Solution {
    public static void rotate(int[] nums, int k) {
        int n = nums.length;
        k %= n;
        reverse(nums, 0, n - 1);
        reverse(nums, 0, k - 1);
        reverse(nums, k, n - 1);

    }
    public static void reverse(int[] nums, int begin, int end){
        while(begin < end){
            int temp = nums[begin];
            nums[begin++] = nums[end];
            nums[end++] = temp;
        }
    }
}

File: 189. Rotate Array.cpp
#include"LeetCodeLibs.h"

class Solution {
public:
    void rotate(vector<int>& nums, int k) {   
        int n = nums.size();
        k %= n;
        reverse(nums.begin(), nums.end());
        reverse(nums.begin(), nums.begin() + k);
        reverse(nums.begin() + k, nums.end());
    }
};

//[1,2,3,4,5,6,7]
//[7,6,5,4,3,2,1]
//[5,6,7,1,2,3,4];


